File: Formularios/Agregar/add_itemManual.cs
using System;
using System.Windows.Forms;

namespace Centrex
{
    public partial class add_itemManual
    {
        // Private esEdicion As Boolean

        private int idUsuario;
        private Guid idUnico;

        public add_itemManual() : base()
        {
            // Esta llamada es exigida por el diseñador.
            InitializeComponent();

            // Agregue cualquier inicialización después de la llamada a InitializeComponent().

        }

        public add_itemManual(ItemEntity i, int _idUsuario, Guid _idUnico) : base()
        {
            // Esta llamada es exigida por el diseñador.
            InitializeComponent();

            // Agregue cualquier inicialización después de la llamada a InitializeComponent().
            txt_item.Text = i.Descript;
            txt_cantidad.Text = i.Cantidad.ToString();
            txt_precio.Text = i.PrecioLista.ToString();
            // esEdicion = True

            idUsuario = _idUsuario;
            idUnico = _idUnico;
        }

        public add_itemManual(int _idUsuario, Guid _idUnico)
        {

            // Esta llamada es exigida por el diseñador.
            InitializeComponent();

            // Agregue cualquier inicialización después de la llamada a InitializeComponent().
            idUsuario = _idUsuario;
            idUnico = _idUnico;
        }

        private void cmd_exit_Click(object sender, EventArgs e)
        {
            Dispose();
        }

        private void cmd_ok_Click(object sender, EventArgs e)
        {
            var cant = default(decimal);
            decimal precio;

            try
            {
                cant = Conversions.ToDecimal(txt_cantidad.Text);
                precio = Conversions.ToDecimal(txt_precio.Text);
            }
            catch (Exception ex)
            {
                Interaction.MsgBox("Valores no válidos: " + ex.Message, MsgBoxStyle.Critical, "Centrex");
                return;
            }


            if (string.IsNullOrEmpty(txt_item.Text))
            {
                Interaction.MsgBox("Debe ingresar un item antes de poder continuar", (MsgBoxStyle)((int)Constants.vbCritical + (int)Constants.vbOKOnly), "ComputronSystems");
                return;
            }
            else if (string.IsNullOrEmpty(txt_cantidad.Text))
            {
                Interaction.MsgBox("Debe ingresar la cantidad antes de poder continuar", (MsgBoxStyle)((int)Constants.vbCritical + (int)Constants.vbOKOnly), "ComputronSystems");
                return;
            }
            else if (Math.Truncate(cant) != cant)
            {
                Interaction.MsgBox("La cantidad no puede ser decimal", (MsgBoxStyle)((int)Constants.vbCritical + (int)Constants.vbOKOnly), "ComputronSystems");
                return;
            }
            else if (string.IsNullOrEmpty(txt_precio.Text))
            {
                Interaction.MsgBox("Debe ingresar el precio antes de poder continuar", (MsgBoxStyle)((int)Constants.vbCritical + (int)Constants.vbOKOnly), "ComputronSystems");
                return;
            }

            var i = new ItemEntity();

            i.Cantidad = Conversions.ToInteger(txt_cantidad.Text);
            i.Descript = txt_item.Text;
            i.PrecioLista = Conversions.ToDecimal(txt_precio.Text);
            i.Descript = "MANUAL";

            Pedidos.AddItemPedidoTmp(i, i.Cantidad ?? 0, Conversions.ToDecimal(i.PrecioLista), idUsuario, idUnico, i.IdTmpPedidoItem > 0 ? i.IdTmpPedidoItem : -1);

            Dispose();
        }

        private void txt_cantidad_KeyPress(object sender, KeyPressEventArgs e)
        {
            if (!generales.esNumero(e))
            {
                e.KeyChar = Conversions.ToChar("");
            }
        }

        private void txt_precio_KeyPress(object sender, KeyPressEventArgs e)
        {
            if (!generales.esNumero(e) & Conversions.ToString(e.KeyChar) != "." & Conversions.ToString(e.KeyChar) != ",")
            {
                e.KeyChar = Conversions.ToChar("");
            }

            if (Strings.Asc(e.KeyChar) == 8)
                return;
        }
    }
}

File: __Viejo/Formularios/Agregar/add_itemManual.vb
Public Class add_itemManual
    'Private esEdicion As Boolean

    Private idUsuario As Integer
    Private idUnico As String

    Public Sub New()
        MyBase.New
        ' Esta llamada es exigida por el diseñador.
        InitializeComponent()

        ' Agregue cualquier inicialización después de la llamada a InitializeComponent().

    End Sub

    Public Sub New(ByVal i As item, ByVal _idUsuario As Integer, ByVal _idUnico As String)
        MyBase.New
        ' Esta llamada es exigida por el diseñador.
        InitializeComponent()

        ' Agregue cualquier inicialización después de la llamada a InitializeComponent().
        txt_item.Text = i.descript
        txt_cantidad.Text = i.cantidad
        txt_precio.Text = i.precio_lista
        'esEdicion = True

        idUsuario = _idUsuario
        idUnico = _idUnico
    End Sub

    Public Sub New(ByVal _idUsuario As Integer, ByVal _idUnico As String)

        ' Esta llamada es exigida por el diseñador.
        InitializeComponent()

        ' Agregue cualquier inicialización después de la llamada a InitializeComponent().
        idUsuario = _idUsuario
        idUnico = _idUnico
    End Sub

    Private Sub cmd_exit_Click(sender As Object, e As EventArgs) Handles cmd_exit.Click
        Me.Dispose()
    End Sub

    Private Sub cmd_ok_Click(sender As Object, e As EventArgs) Handles cmd_ok.Click
        Dim cant As Decimal
        Dim precio As Decimal

        Try
            cant = txt_cantidad.Text
            precio = txt_precio.Text
        Catch ex As Exception
        End Try


        If txt_item.Text = "" Then
            MsgBox("Debe ingresar un item antes de poder continuar", vbCritical + vbOKOnly, "ComputronSystems")
            Exit Sub
        ElseIf txt_cantidad.Text = "" Then
            MsgBox("Debe ingresar la cantidad antes de poder continuar", vbCritical + vbOKOnly, "ComputronSystems")
            Exit Sub
        ElseIf Math.Truncate(cant) <> cant Then
            MsgBox("La cantidad no puede ser decimal", vbCritical + vbOKOnly, "ComputronSystems")
            Exit Sub
        ElseIf txt_precio.Text = "" Then
            MsgBox("Debe ingresar el precio antes de poder continuar", vbCritical + vbOKOnly, "ComputronSystems")
            Exit Sub
        End If

        Dim i As New item

        With i
            .cantidad = txt_cantidad.Text
            .descript = txt_item.Text
            .precio_lista = txt_precio.Text
            .item = "MANUAL"
        End With

        addItemPedidotmp(i, i.cantidad, i.precio_lista, idUsuario, idUnico, edita_item.id_item_temporal)

        Me.Dispose()
    End Sub

    Private Sub txt_cantidad_KeyPress(sender As Object, e As KeyPressEventArgs) Handles txt_cantidad.KeyPress
        If Not esNumero(e) Then
            e.KeyChar = ""
        End If
    End Sub

    Private Sub txt_precio_KeyPress(sender As Object, e As KeyPressEventArgs) Handles txt_precio.KeyPress
        If Not esNumero(e) And (e.KeyChar <> "." And e.KeyChar <> ",") Then
            e.KeyChar = ""
        End If

        If Asc(e.KeyChar) = 8 Then Exit Sub
        e.KeyChar = isDecimalOk(e)
    End Sub
End Class